controllers/spannerautoscaleschedule_controller.go

```diff
@@ -0,0 +1,136 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlreconcile "sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+	spannerv1beta1 "github.com/mercari/spanner-autoscaler/api/v1beta1"
+)
+
+// SpannerAutoscaleScheduleReconciler reconciles a SpannerAutoscaleSchedule object
+type SpannerAutoscaleScheduleReconciler struct {
+	ctrlClient ctrlclient.Client
+	scheme     *runtime.Scheme
+	log        logr.Logger
+}
+
+type SpannerAutoscaleScheduleReconcilerOption interface {
+	applySpannerAutoscaleScheduleReconciler(r *SpannerAutoscaleScheduleReconciler)
+}
+
+func (o withLog) applySpannerAutoscaleScheduleReconciler(r *SpannerAutoscaleScheduleReconciler) {
+	r.log = o.logger.WithName("schedule")
+}
+
+func NewSpannerAutoscaleScheduleReconciler(
+	ctrlClient ctrlclient.Client,
+	scheme *runtime.Scheme,
+	opts ...Option,
+) *SpannerAutoscaleScheduleReconciler {
+	r := &SpannerAutoscaleScheduleReconciler{
+		ctrlClient: ctrlClient,
+		scheme:     scheme,
+	}
+
+	for _, option := range opts {
+		if opt, ok := option.(SpannerAutoscaleScheduleReconcilerOption); ok {
+			opt.applySpannerAutoscaleScheduleReconciler(r)
+		}
+	}
+
+	return r
+}
+
+//+kubebuilder:rbac:groups=spanner.mercari.com,resources=spannerautoscaleschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=spanner.mercari.com,resources=spannerautoscaleschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=spanner.mercari.com,resources=spannerautoscaleschedules/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups=spanner.mercari.com,resources=spannerautoscaler/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *SpannerAutoscaleScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	nn := req.NamespacedName
+	log := r.log.WithValues("namespaced-name", nn, "emitter", "schedule")
+
+	var sas spannerv1beta1.SpannerAutoscaleSchedule
+	if err := r.ctrlClient.Get(ctx, nn, &sas); err != nil {
+		err = ctrlclient.IgnoreNotFound(err)
+		if err != nil {
+			log.Error(err, "failed to get spanner-autoscale-schedule")
+			return ctrlreconcile.Result{}, err
+		}
+
+		return ctrlreconcile.Result{}, nil
+	}
+
+	var sa spannerv1beta1.SpannerAutoscaler
+	nnsa := types.NamespacedName{
+		Name:      sas.Spec.TargetResource,
+		Namespace: nn.Namespace,
+	}
+	if err := r.ctrlClient.Get(ctx, nnsa, &sa); err != nil {
+		err = ctrlclient.IgnoreNotFound(err)
+		if err != nil {
+			log.Error(err, "failed to get spanner-autoscaler")
+			return ctrlreconcile.Result{}, err
+		}
+
+		return ctrlreconcile.Result{}, nil
+	}
+
+	log.Info("registering schedule with spanner-autoscaler", "schedule", sas.Name, "autoscaler", sa.Name)
+	log.V(1).Info("registering schedule with spanner-autoscaler", "schedule", sas, "autoscaler", sa)
```

Comment:
One of these logs seems to be good enough.

Reply:
I kept two because `Debug` level logs the full objects, which can be useful for finding issues during local development etc. And at the `Info` level in production, just the names of the objects should be sufficient to confirm that the newly created `schedule` was successfully `registered` with the corresponding autoscaler. No extra clutter there.

I have now removed the `Debug` level log.

Reply:
I see, thank you for your comment!